app/util/logs/index.test.ts

```diff
@@ -14,6 +14,8 @@ jest.mock('../../core/Engine', () => ({
 describe('logs :: generateStateLogs', () => {
   it('should generate the state logs correctly without the explicitly deleted controller states', async () => {
     const mockStateInput = {
+      appVersion: 1,
+      buildNumber: 123,
       engine: {
```

Comment:
to make this test more clear about the fact that the version and build infos are not part of the state initially, and remove possible confusion, you can remove values from mock state and add them at generateStateLog call time like you do on the actual code:
```suggestion
    const mockStateInput = {
      engine: {
```

Reply:
Test case added ! Great point!